tools/IoTEdgeDevOps/DevOpsLib/BugManagement.cs

```diff
@@ -26,37 +26,75 @@ public BugManagement(DevOpsAccessSetting accessSetting)
             IFlurlRequest workItemQueryRequest = ((Url)requestPath)
                 .WithBasicAuth(string.Empty, this.accessSetting.PersonalAccessToken)
-                .SetQueryParam("api-version", "5.1");
+                .WithHeader("Content-Type", "application/json-patch+json")
+                .SetQueryParam("api-version", "6.0");
+
+            var jsonBody = new object[]
+            {
+                new {
+                    op = "add",
+                    path = "/fields/System.Title",
+                    from = string.Empty,
+                    value = $"Test failure on {branch}: {build.DefinitionId.ToString()} {build.BuildNumber}"
+                },
+                new
+                {
+                    op = "add",
+                    path = "/fields/Microsoft.VSTS.TCM.ReproSteps",
+                    from = string.Empty,
+                    value = $"This bug is autogenerated by the vsts-pipeline-sync service. Link to failing build:<div> {build.WebUri}"
+                },
+                new
+                {
+                    op = "add",
+                    path = "/fields/Microsoft.VSTS.Common.Priority",
+                    from = string.Empty,
+                    value = "0"
+                },
+                new
+                {
+                    op = "add",
+                    path = "/fields/System.AreaPath",
+                    from = string.Empty,
+                    value = "One\\IoT\\Platform\\IoTEdge"
+                },
+                new
+                {
+                    op = "add",
+                    path = "/relations/-",
+                    value = new
+                    {
+                        rel = "Hyperlink",
+                        url = $"{build.WebUri}"
+                    }
+                }
+            };
 
             JObject result;
             try
             {
                 IFlurlResponse response = await workItemQueryRequest
-                    .PostJsonAsync(new { query = bugQuery.GetWiqlFromConfiguration() });
+                    .PostJsonAsync(jsonBody);
 
                 result = await response.GetJsonAsync<JObject>();
             }
             catch (FlurlHttpException e)
             {
-                Console.WriteLine($"Failed making call to vsts work item api: {e.Message}");
+                string message = $"Failed making call to vsts work item api: {e.Message}";
+                Console.WriteLine(message);
                 Console.WriteLine(e.Call.RequestBody);
                 Console.WriteLine(e.Call.Response.StatusCode);
                 Console.WriteLine(e.Call.Response.ResponseMessage);
-                return 0;
-            }
 
-            if (!result.ContainsKey("queryType"))
-            {
-                return 0;
+                throw new Exception(message);
```

Comment:
I am rethrowing as I want this func to make it clear that something has gone wrong. The only success case for this function call should be if a bug is successfully made.